polaris-react/src/components/Link/Link.tsx

```diff
@@ -40,8 +40,7 @@ export function Link({
 }: LinkProps) {
   const i18n = useI18n();
   let childrenMarkup = children;
-
-  if (external && typeof children === 'string') {
+  if (external && typeof children !== 'object') {
```

Comment:
Can we just check `external` here and remove `&& typeof children !== 'object'`?

Reply:
I thought about this. The reason we do this is because if a react node is passed in we don't know what content will render or if we should append an external icon. It's a bit weird... 

I would be happy removing this check in a breaking change as it could significantly change some user interfaces.